neurox/interpretation/probeless.py

```diff
@@ -0,0 +1,188 @@
+
+def _get_tag_wise_ranking(mean_vectors_by_label, tag):
+
+    qz = mean_vectors_by_label[tag]
+    summation = np.abs(np.subtract(qz, qz))
+
+    for c, qzz in mean_vectors_by_label.items():
+        
+        if (tag != c):
+            summation = np.add(summation, np.abs(np.subtract(qz,qzz)))
+    
+    ranking = np.argsort(summation)[::-1].tolist()
+    return summation, ranking 
+
+
+def _get_layer_vector(X_t, y_t, NUM_LAYERS, layer=-1):
+
+    if (layer == -1):
+        layer_X_t = X_t
+
+    elif (layer <= NUM_LAYERS):
+        layer_X_t = ablation.filter_activations_by_layers(X_t, [layer], NUM_LAYERS)
+        
+    else:
+        raise ValueError(f"Invalid layer number {layer}")
+
+    return layer_X_t
+
+def get_top_overall_neurons(X_t, y_t, NUM_LAYERS, layer=-1):
+
+    """
+    Returns a list of top neurons w.r.t the overall task e.g. POS
+
+    Parameters
+    ----------
+    X_t : numpy.ndarray
+    Numpy Matrix of size [``NUM_TOKENS`` x ``NUM_NEURONS``]. Usually the 
+    output of ``interpretation.utils.create_tensors``
+    y_t : numpy.ndarray
+    Numpy Vector of size [``NUM_TOKENS``] with class labels for each input
+    token. Usually the output of ``interpretation.utils.create_tensors``.
+    NUM_LAYERS : int
+    Total number of layers in the network
+    layer : int, optional
+    layer number for which ranking needs to be extracted, -1 for entire network
+
+
+    Returns
+    -------
+    ranking : list
+
+    """
+    
+    layer_X_t = _get_layer_vector(X_t, y_t, NUM_LAYERS, layer)
+    avg_embeddings, average_embeddings_by_label = _get_mean_vectors(layer_X_t, y_t)
+    ranking = _get_overall_ranking(avg_embeddings)
+    
+    return ranking
+
+def get_top_neurons_for_tag(X_t, y_t , tag, NUM_LAYERS, layer=-1):
+
+    """
+    Returns a list of top neurons w.r.t a tag e.g. noun
+
+    Parameters
+    ----------
+    X_t : numpy.ndarray
+    Numpy Matrix of size [``NUM_TOKENS`` x ``NUM_NEURONS``]. Usually the 
+    output of ``interpretation.utils.create_tensors``
+    y_t : numpy.ndarray
+    Numpy Vector of size [``NUM_TOKENS``] with class labels for each input
+    token. Usually the output of ``interpretation.utils.create_tensors``.
+    tag : int 
+    tag for which rankings are extracted
+    NUM_LAYERS : int
+    Total number of layers in the network
+    layer : int, optional
+    layer number for which ranking needs to be extracted, -1 for entire network
+
+    Returns
+    -------
+    ranking : list
+
+    """
+
+    layer_X_t = _get_layer_vector(X_t, y_t, NUM_LAYERS, layer)
+    avg_embeddings, average_embeddings_by_label = _get_mean_vectors(layer_X_t, y_t)
+    summation, ranking = _get_tag_wise_ranking(average_embeddings_by_label, tag)
+
+    return ranking
+
+def get_top_neurons_for_all_tags(X_t, y_t, NUM_LAYERS, layer=-1):
+
+
+    """
+    Returns a dictionary of tags along with top neurons for each tag
+    Returns a list of overall ranking
+
+    Parameters
+    ----------
+    X_t : numpy.ndarray
```

Comment:
If X_t is referening to train, rename to `X_train`, if it can be any tensor in general, rename to `X`; same for `y_t`